AVFoundation-Combine/Video Player/VideoPlayerViewController.swift

```diff
@@ -0,0 +1,163 @@
+//
+
+import UIKit
+import AVKit
+
+import Combine
+
+final class VideoPlayerViewController: UIViewController {
+    
+    /// A sample video URL
+    ///
+    /// **Big Buck Bunny (2008)**
+    ///
+    /// A recently awoken enormous and utterly adorable fluffy rabbit is heartlessly harassed by a flying squirrel's gang of rodents who are determined to squash his happiness.
+    private let videoURL = URL(string: "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4")!
+    
+    /// A `Set` to store all our `Publisher` susbcriptions
+    private var subscriptions = Set<AnyCancellable>()
+    
+    /// A flag to keep track of whether the user is using `progressSlider` to scrub trough the video timeline. Used to prevent the thumb in the slider from jumping back and forth while `seek` is in progress.
+    private var isProgressSliderScrubbing: Bool = false
+    
+    lazy private var customUI: VideoPlayerView = {
+        VideoPlayerView()
+    }()
+    
+    lazy private var avPlayerViewController: AVPlayerViewController = {
+        AVPlayerViewController()
+    }()
+    
+    // MARK: UI setup
+    
+    private func setupUI() {
+        addChild(avPlayerViewController)
+        avPlayerViewController.entersFullScreenWhenPlaybackBegins = true
+        view.addSubviewAndFillBounds(avPlayerViewController.view)
+        avPlayerViewController.didMove(toParent: self)
+        avPlayerViewController.view.backgroundColor = UIColor(named: "Background")
+        avPlayerViewController.showsPlaybackControls = false
+        guard let contentOverlayView = avPlayerViewController.contentOverlayView else {
+            fatalError("`contentOverlayView` is required.")
+        }
+        contentOverlayView.addSubviewAndFillBounds(customUI)
+        customUI.translatesAutoresizingMaskIntoConstraints = false
+        customUI.playbackButton.addTarget(self, action: #selector(togglePlayback), for: .touchUpInside)
+        customUI.progressSlider.addTarget(self, action: #selector(onSliderThumbTouchedDown), for: .touchDown)
+        customUI.progressSlider.addTarget(self, action: #selector(onSliderThumbTouchedUp), for: .touchUpOutside)
+        customUI.progressSlider.addTarget(self, action: #selector(onSliderThumbTouchedUp), for: .touchUpInside)
+    }
+    
+    // MARK: UI Actions
+    
+    // MARK: Scubber
+    
+    @objc private func onSliderThumbTouchedDown() {
+        isProgressSliderScrubbing = true
+    }
+    
+    @objc private func onSliderThumbTouchedUp() {
+        avPlayerViewController.player?.seek(to: CMTime(seconds: Double(customUI.progressSlider.value), preferredTimescale: 1)) {[weak self] _ in
+            self?.isProgressSliderScrubbing = false
+        }
+    }
+    
+    // MARK: Play / Pause button
+    
+    @objc private func togglePlayback() {
+        avPlayerViewController.player?.rate == 0.0 ? avPlayerViewController.player?.play() : avPlayerViewController.player?.pause()
```

Comment:
You already leverage `ratePublisher()` a few lines in the setup. It might be better to just bind that value to a bool property on this view controller like so:
```swift
private var isPlaying = false
// ... in `setup(_:)`
rateStream
    .receive(on: DispatchQueue.main)
    .removeDuplicates()
    .map { $0 == 0 ? false : true }
    .assign(to: \.isPlaying, on: self)
    .store(in: &subscriptions)
```